### PR TITLE
Cache the `backportPrExists` function

### PR DESCRIPTION
**Pull request by yardenshoham**
To avoid calling the github API so much

### CODE REVIEW

File: src/github_test.ts [hunk 1]
```diff
@@ -140,3 +141,23 @@ Deno.test("fetchLastComment() returns the appropriate comment", async () => {
...
+    Object.entries(prAndVersionToBackportExists).map(
+      async ([prAndVersion, backportExists]) => {
+        const [prNumber, version] = prAndVersion.split("_");
+        const result = await backportPrExists(
+          { number: Number(prNumber) },
+          version,
+        );
+        assertEquals(result, backportExists);
+      },
+    ),
```
Comment (silverwind) [inline src/github_test.ts:161]:
```suggestion
    Object.entries(prAndVersionToBackportExists).map(async ([prAndVersion, backportExists]) => {
      const [prNumber, version] = prAndVersion.split("_");
      const result = await backportPrExists(
        { number: Number(prNumber) },
        version,
      );
      assertEquals(result, backportExists);
    }),
```

prefer to put callback signature on same line, 97 chars line length is still ok with me.

**Comment by silverwind** [comment]
We can't persist this cache somewhere, right? How often does the script run, btw?

**Comment by silverwind** [comment]
Could also memoize the `backportPrExists` function with something like https://github.com/aduth/memize, would be less code overall.

**Comment by yardenshoham** [comment]
> We can't persist this cache somewhere, right? How often does the script run, btw?
The cache will persist until the next deployment/cache

> Could also memoize the `backportPrExists` function with something like https://github.com/aduth/memize, would be less code overall.
I will try

**Comment by yardenshoham** [comment]
Actually we don't want to memoize when it returns false so I will revert

**Comment by silverwind** [comment]
Right, if you only want to cache one specific return value, then its unsuitable.

File: src/github.ts [hunk 2]
```diff
@@ -257,16 +264,23 @@ export const backportPrExists = async (
...
+    return true;
+  }
 
   // also check if a branch that looks like the backport branch (getPrBranchName) exists
   response = await fetch(
     `${GITHUB_API}/repos/${Deno.env.get("BACKPORTER_GITEA_FORK")}/branches/${
       getPrBranchName(pr.number, giteaMajorMinorVersion)
     }`,
-    { headers: HEADERS },
+    { headers: HEADERS, method: "HEAD" },
```
Comment (silverwind) [inline src/github.ts:277]: HEAD method counts the same as GET for [the rate limit](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#calculating-points-for-the-secondary-rate-limit), but it's still a good change to keep.

**Comment by silverwind** [comment]
> The cache will persist until the next deployment/cache

So it's continously running? I have no idea how this works 😆

File: src/github_test.ts (hunk 1 above)
Comment (silverwind) [reply src/github_test.ts:161]: Why not commit? Do you have some code formatter ongoing?

**Comment by yardenshoham** [comment]
Yeah. When I was hosting it I used deno deploy but now it runs as an app on fly.io

File: src/github_test.ts (hunk 1 above)
Comment (yardenshoham) [reply src/github_test.ts:161]: I committed it https://github.com/GiteaBot/gitea-backporter/pull/128/commits/e8187bdb9dff59dd822d78b724c00f36f57e55fd then ran deno fmt https://github.com/GiteaBot/gitea-backporter/pull/128/commits/ed9ee2ab7d62fff89c4d9326ecb97ec9f65cf3ea

Comment (silverwind) [reply src/github_test.ts:161]: I see, so formatting nits are basically auto-rejected :D

**Review by silverwind — APPROVED** [review]

**Pull request merged**